### PR TITLE
[receiver/postgresql] Use Resource Attributes

### LINKED ISSUE #12960: [receiver/postgresql] Fix Collection To Use Resource Attributes
**Is your feature request related to a problem? Please describe.**
According to the [resource specification]( https://github.com/open-telemetry/opentelemetry-specification/blob/141a3ef0bf1eba0b6d260335bbe0ce7af9387cfc/specification/metrics/data-model.md#resource-attributes-1), the receiver metrics should not specify identifying information such as tables or databases on the metrics. The metric attributes `table` and `database` will be moved to resource attributes `postgresql.table` and `postgresql.database` respectively.

**Describe the solution you'd like**
Fix current metrics to use resource attributes.

**Additional context**
Initial issue found in #12338

### PR DESCRIPTION
**Pull request by schmikei**
**Description:** <Describe what has changed.>
Changes collection to use resource attributes to identify the metrics rather than metric attributes. This required some changes to the client and scraper.

**Link to tracking Issue:** Resolves #12960 

**Testing:** Scraper Test and Integration Testing

**Documentation:** Changelog entry for the breaking change.

### CODE REVIEW

**Comment by schmikei** [comment]
@djaglowski this is the breaking change for resource attributes for the postgresqlreceiver. Do you mind taking a look?

File: receiver/postgresqlreceiver/client.go [hunk 1]
```diff
@@ -17,24 +17,23 @@ package postgresqlreceiver // import "github.com/open-telemetry/opentelemetry-co
...
 )
 
 type client interface {
 	Close() error
-	getCommitsAndRollbacks(ctx context.Context, databases []string) ([]MetricStat, error)
-	getBackends(ctx context.Context, databases []string) ([]MetricStat, error)
-	getDatabaseSize(ctx context.Context, databases []string) ([]MetricStat, error)
-	getDatabaseTableMetrics(ctx context.Context) ([]MetricStat, error)
-	getBlocksReadByTable(ctx context.Context) ([]MetricStat, error)
+	getDatabaseStats(ctx context.Context, databases []string) (map[string]databaseStats, error)
```
Comment (djaglowski) [inline receiver/postgresqlreceiver/client.go:32]: At a high level, the purpose of this PR is to address the way we represent data. I don't quite understand why that necessarily has anything to do with retrieving the data. Is it possible there are unrelated changes here? If not, what's the motivation for reworking the client in this PR?

Comment (schmikei) [reply receiver/postgresqlreceiver/client.go:32]: The main motivation is that `EmitForResource()` is order dependent so we need a way to consistently record datapoints that belong to the correct resource in an organized manner. 

For example if you collect all the database size datapoints you cannot emit them by iterating over that `MetricStat` slice because it would contain database size datapoints for both `database1`, `database2`, etc. This motivated the client to provide lookup tables so we can refer to the same database without having to go through another lookup/reducing to organize the datapoints.

**Review by djaglowski — CHANGES_REQUESTED** [review]
@schmikei, I cannot follow all of these changes in one PR. Please find a way to break this up into smaller PRs.

If you need to rework the client, consider doing that as a standalone PR. There should be no changes to the output of the scraper. Try to make only necessary changes.

**Comment by schmikei** [comment]
@djaglowski since #13084 was merged in, do you mind taking a re-look at this change that focuses on feature gating resource attributes?

File: receiver/postgresqlreceiver/metadata.yaml [hunk 2]
```diff
@@ -1,20 +1,37 @@
 name: postgresqlreceiver
 
+resource_attributes:
+  postgresql.database:
+    description: The name of the database.
+    type: string
+  postgresql.table:
```
Comment (djaglowski) [inline receiver/postgresqlreceiver/metadata.yaml:7]: These should probably have `.name` on the end, so that it's possible to reuse the namespaces (eg. `postgresql.table.id`)

File: receiver/postgresqlreceiver/scraper_test.go [hunk 3]
```diff
@@ -92,6 +92,44 @@ func TestScraperNoDatabaseMultiple(t *testing.T) {
 	require.NoError(t, scrapertest.CompareMetrics(expectedMetrics, actualMetrics))
 }
 
+func TestScraperWithResourceAttributeFeatureGate(t *testing.T) {
```
Comment (djaglowski) [inline receiver/postgresqlreceiver/scraper_test.go:95]: Can we add a new case to `integration_test.go` as well?

**Review by djaglowski — COMMENTED** [review]
Looks good overall. Just a couple minor things.

File: receiver/postgresqlreceiver/scraper_test.go (hunk 3 above)
Comment (schmikei) [reply receiver/postgresqlreceiver/scraper_test.go:95]: Yeah sure I can look into it. Looked into it initially and wasn't sure how to do it with the featuregate Registry while not affecting the other integration test. Do you happen to know of an example or if it could be solved using  the `	featuregate.GetRegistry().MustApply()` function?

Comment (schmikei) [reply receiver/postgresqlreceiver/scraper_test.go:95]: Testing around and I think the `featuregate.GetRegistry().MustApply()` should work 👍 probably should make sure that it cleans up the featuregate registry before that portion of the integration test completes.

Comment (schmikei) [reply receiver/postgresqlreceiver/scraper_test.go:95]: Just pushed an attempt at making an integration test let me know if that works!

Comment (djaglowski) [reply receiver/postgresqlreceiver/scraper_test.go:95]: Thanks @schmikei. I think that looks like it'll work.

**Comment by djaglowski** [comment]
I believe tests are failing due to adding `.name` to the new attributes, but not updating the expected files.

**Review by djaglowski — APPROVED** [review]
